docker-action-test/Dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM --platform=$BUILDPLATFORM docker.io/golang:1.21-alpine3.18 as builder
```

Comment:
can we submit this and the `dockerfile_workflow_test.yaml` as a separate PR so we can merge them in and then this PR will actually test against them.